eth/exceptions.py

```diff
@@ -33,6 +33,7 @@ class HeaderNotFound(PyEVMError):
 class BlockNotFound(PyEVMError):
     """
     Raised when the block with the given number/hash does not exist.
+    This might happen if a header is present, but the transactions or uncles are not.
```

Comment:
Maybe clearer to say:

```suggestion
    This will happen, for example, if the transactions or uncles are
    not saved in the database.
```